lib/matplotlib/cbook/__init__.py

```diff
@@ -1326,7 +1326,13 @@ def _check_1d(x):
         return np.atleast_1d(x)
     else:
         try:
-            x[:, None]
+            ndim = x[:, None].ndim
```

Comment:
you don't want to do a `np.asarray` here first? That should make it work with older pandas versions, but also robust to potential future changes (eg if pandas starts raising for that)

Reply:
we do not want to `asarray` things we don't have to (in particular because in some cases that can strip units off of things).  So long as future versions raises one of the two exceptions caught below it will "just work" and as soon as you decide what exception you are going to raise we can add it.

The point of this is to check if we _need_ to `asarray` (which in done implicitly in `atleast_1d`).

Reply:
OK, I see. I suppose if we raise an error it will be an IndexError, so that would already be covered then